Bottom: ac565af90cf93842f73337a8d5ed0d71ade455e1
Top:    7a7a61cb5f035bf471de130036d571d9f26b3a1d
Author: melsdoerfer <michael@elsdoerfer.com>
Date:   2009-11-04 08:46:41 +0100

No longer convert an item id (guid) into an absolute link if isPermaLink=="false".

While not strictly necessary to run FeedPlatform, it is strongly recommended. For example, one significant issue arising from this is that the domain name used to build the absolute url may change (say the feed is redirected to a new address), thus effectively changing all the ids.

Reported as issue #107.


---

diff --git a/feedparser/feedparser.py b/feedparser/feedparser.py
index 5c075a3..4e94bb0 100644
--- a/feedparser/feedparser.py
+++ b/feedparser/feedparser.py
@@ -732,7 +732,9 @@ class _FeedParserMixin:
                 
         # resolve relative URIs
         if (element in self.can_be_relative_uri) and output:
-            output = self.resolveURI(output)
+            # but not for isPermaLink=False guids in rss
+            if not (element == 'id' and self.version.startswith('rss') and self.inentry and not self.guidislink):
+                output = self.resolveURI(output)
         
         # decode entities within embedded markup
         if not self.contentparams.get('base64', 0):
